### PR TITLE
luminous: 'radosgw-admin sync error trim' only trims partially

### PR DESCRIPTION
**Pull request by smithfarm**
http://tracker.ceph.com/issues/24983

### CODE REVIEW

**Comment by smithfarm** [comment]
```
Scanning dependencies of target unittest_rgw_crypto
[ 98%] Building CXX object src/test/rgw/CMakeFiles/unittest_rgw_crypto.dir/test_rgw_crypto.cc.o
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_admin.cc: In function ‘int bucket_source_sync_status(RGWRados*, const RGWZone&, const RGWZone&, RGWRESTConn*, const RGWBucketInfo&, int, std::ostream&)’:
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_admin.cc:2327:38: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (size_t shard_id = 0; shard_id < total_shards; shard_id++) {
                                      ^
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_admin.cc: In function ‘int trim_sync_error_log(int, const real_time&, const real_time&, const string&, const string&, int)’:
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_admin.cc:2519:12: error: ‘std::this_thread’ has not been declared
       std::this_thread::sleep_for(std::chrono::milliseconds(delay_ms));
            ^
src/rgw/CMakeFiles/radosgw-admin.dir/build.make:62: recipe for target 'src/rgw/CMakeFiles/radosgw-admin.dir/rgw_admin.cc.o' failed
make[3]: *** [src/rgw/CMakeFiles/radosgw-admin.dir/rgw_admin.cc.o] Error 1
```

**Comment by smithfarm** [comment]
@theanalyst @cbodley What do I use in place of `std::this_thread` ?

```
src/rgw/rgw_admin.cc:2519:12: error: ‘std::this_thread’ has not been declared
```

File: src/rgw/rgw_admin.cc [hunk 1]
```diff
@@ -2498,6 +2498,30 @@ int create_new_bucket_instance(RGWRados *store,
...
+    int ret = store->time_log_trim(oid, start_time, end_time,
+                                   start_marker, end_marker);
+    if (ret == -ENODATA) {
+      return 0;
+    }
+    if (ret < 0) {
+      return ret;
+    }
+    if (delay_ms) {
+      std::this_thread::sleep_for(std::chrono::milliseconds(delay_ms));
```
Comment (smithfarm) [inline src/rgw/rgw_admin.cc]: Problem with `std::this_thread` here.

**Comment by cbodley** [comment]
> @theanalyst @cbodley What do I use in place of `std::this_thread` ?

this is available in c++11, we're probably just missing an `#include <thread>` in rgw_admin.cc. i'm guessing that mimic/master are getting that include from some other header

**Comment by smithfarm** [comment]
Thanks @cbodley 

I added `#include <thread>` for `std::this_thread` and `#include <chrono>` for `std::chrono::milliseconds`

**Comment by yuriw** [comment]
[wip-yuri3-testing-2018-09-17-1952-luminous](https://trello.com/c/tDY6qyzl/407-wip-yuri3-testing-2018-09-17-1952-luminous)

**Pull request merged**